library/src/morpheus-master/js/src/sakai.morpheus.connection-manager.js

```diff
@@ -1,6 +1,6 @@
 (function ($) {
 
-    if (!portal.loggedIn){
+    if (!portal.loggedIn || !(portal.bullhorns && portal.bullhorns.enabled)){
```

Comment:
I thought we made i18n properties web endpoint public not sure we need this anymore.

Reply:
The point of this is not to avoid doing the call if it is not public, the point is to avoid to do the call if it is unnecessary.

If you are not logged in and / or bullhorns are disabled the ajax call is useless as long as all this JS is useless because data got by AJAX is never used.

In our servers we have bullhorns disabled and this call is made between 10 and 16 million times a month. 16 millions of useless AJAX calls :)

On !gateway is unnecesary too because you will never be notified by any bullhorns if you are not logged in so this optimize.

Reply:
I understand what your saying about the client continuing to make these ajax calls. But if more tools start fetching i18n strings via this web service how do we propose to handle those?

What your proposing is to continually update this js with those tools/features?

Personally I feel that the ajax requests that continue to ping sakai after a user has logged out or become inactive is just incorrect. I recall seeing courier service and gradebookng doing this which is just seems wrong. I do like the use of the browsers local storage for replacing a call for this type of data so kudos on that. However I think this problem needs a better long term solution.

I am not against this as a temporary fix, just advocating for a longer term solution.

Reply:
@ern I think we can't refactor all bullhorns / connection manage in this way. We don't know what were the decision made about this ajax calls. I agree with you, this tool needs to be refactored but this is beyond this JIRA. Feel free to create a new one and we can discuss with our bosses if we need to do that job.